1_nldas_cubes/code/nccopy_utils.R

```diff
@@ -0,0 +1,45 @@
+nccopy_nldas <- function(nc_filepath){
+  
+  nc_filename <- basename(nc_filepath)
+  
+  nldas_url <- nldas_url_from_file(nc_filename)
+  message(nldas_url)
+  output <- system(sprintf("nccopy -m 15m %s %s", nldas_url, nc_filepath))
+  
+  if(output){
+    unlink(nc_filename)
+    stop(nldas_url, ' **FAILED nccopy**', call. = FALSE)
+  }
+  # check/verify file has the appropriate number of timesteps...fail if not
+  expected_steps <- length(nldad_steps_from_file(nc_filename))
+  nc <- ncdf4::nc_open( nc_filepath )
+  nc_steps <- nc$dim$time$len
+  if (nc_steps != expected_steps){
+    ncdf4::nc_close(nc)
+    unlink(nc_filename)
+    stop('incomplete file ', nc_filename , '\nexpected ', 
+         expected_steps, ' steps, found ', nc_steps, ' steps\nfor ', nldas_url, call. = FALSE)
+  }
+  invisible()
+}
+
+nldad_steps_from_file <- function(nc_filename){
+  file_chunks <- strsplit(nc_filename,'[_]')[[1]]
+  time_range <- strsplit(file_chunks[2],'[.]')[[1]] %>% as.numeric
+  return(seq(time_range[1], time_range[2]))
+}
+
+nldas_url_from_file <- function(nc_filename){
+  file_chunks <- strsplit(nc_filename,'[_]')[[1]]
+  lat_i <- write_grid(file_chunks[3])
+  lon_i <- write_grid(file_chunks[4])
+  time_i <- write_grid(file_chunks[2])
+  var <- strsplit(file_chunks[5],'[.]')[[1]][1]
+  sprintf('http://hydro1.sci.gsfc.nasa.gov/dods/NLDAS_FORA0125_H.002?lon%s,time%s,lat%s,%s%s%s%s', lon_i, time_i, lat_i, var, time_i, lat_i, lon_i)
+}
```

Comment:
this name parsing function seems like a pretty efficient approach for this use case. but just to plant another possibility in your head for future projects, another way you could do something like this is to 
1) create a data.frame of all the URLs you want to pull from, where URL is in one column and an informative target name is in another column. in this case, the informative target name could be exactly the target names you're already constructing (in https://github.com/USGS-R/lake-temperature-model-prep/pull/3/files?utf8=%E2%9C%93&diff=split#diff-be5d8c0a755e39f42bb3feeb0b9ba258R118). create that data.frame in the master remake file (here 0_config_rmk.yml) and assign it to an R object or file...for this example call it `lookup_table`.
2) then, rather than also writing out a function with logic to parse in the opposite direction (here, going from file=target back to URL), you could just look up the URL in `lookup_table`.

for this to work, the task-step command would look something like this:
```
  1_nldas_cubes/out/NLDAS_7.12_223.223_141.141_dlwrfsfc.nc:		
    command: nccopy_nldas(target_name, lookup_table)
```
and `lookup_table` would need to be a target in the master remake file, and the master remake file would need to be included in the task remake file (you're already doing this last bit).

I'm going back and forth on whether lookup_table needs to get wrapped in `I('')` and retrieved away from remake's prying gaze (through a `get(lookup_table)` or `readRDS(lookup_table)` within `nccopy_nldas()`)... but I think as long as the lookup_table and the tasks in this task table are both changed anytime one of them is changed, it shouldn't matter.

Reply:
Yes - I like this idea. I probably won't pivot for this cube fetcher file naming/lookup right now, but this seems like the appropriate way to deal with the challenge of naming files intuitively that contain a bunch of different WQP sites. @limnoliver 👆 a look up table (or list) like this for file, params, sites might be a good thing to look into.